src/controllers/orderControllers.ts

```diff
@@ -0,0 +1,108 @@
+import { Request, Response } from "express";
+import Product from "../models/Product";
+import Order from "../models/Order";
+
+export const getAllOrders = async (req: Request, res: Response) => {
+  try {
+    const loggedUser: any = req.user;
+    const userId: string = loggedUser.dataValues.id as string;
+    const userRole = loggedUser.dataValues.Role;
+    const roleName = userRole.dataValues.name;
+    if (roleName === "buyer") {
+      const oreders = await Order.findAll({
+        where: { buyerId: loggedUser.dataValues.id },
+        include: [
+          {
+            model: Product,
+            attributes: ["productName", "productPrice"]
+          }
+        ]
+      });
+      return res.status(200).json({ oreders });
+    }
+    if (roleName === "seller") {
+      const data = await Order.findAll({
+        include: [
```

Comment:
Something is wrong in this query. Are you fetching all orders?

Reply:
once you logged in as a seller it return all orders for specific products of that seller and when you logged in as a buyer it return all orders related to you